### PR TITLE
vf-3 -implement role-based access control and enforcement

### PR DESCRIPTION
**Pull request by asterns**

### CODE REVIEW

File: auth0/sync-roles.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  getTwoLeggedToken, 
+  getAuth0Roles,
+  addAuth0Role,
+  updateAuth0Role,
+} = require('../server/security');
+const roles = getDenormalizedRoles();
+getTwoLeggedToken({
+  client_id: process.env.AUTH0_DEPLOYMENT_CLIENT_ID, 
+  client_secret: process.env.AUTH0_DEPLOYMENT_SECRET, 
+  audience: "https://asterns.auth0.com/api/v2/",
```
Comment (sjforman) [inline auth0/sync-roles.js]: Should be an environment variable?

File: config/config.js [hunk 2]
```diff
@@ -16,6 +17,22 @@ convict.addFormat({
...
+  convict.addFormat({
+    name: 'role-array',
+    validate: function(roles, schema) {
+      if (!Array.isArray(roles)) {
+        throw new Error('must be of type Array');
+      }
+      const roleMap = {};
+      for (role of roles) {
+        if (roleMap[role.slug]) {
+          throw new Error(`${partner.slug} defined more than once`);
```
Comment (sjforman) [inline config/config.js]: `partner.slug` here should be, I think, `role.slug`

**Comment by sjforman** [comment]
(Also, to confirm, I was able to sync roles to auth0, tests all pass, and confirmed that, after I gave myself a role in auth0, I was able to see that role in the decoded jwt).

File: auth0/sync-roles.js (hunk 1 above)
Comment (asterns) [reply auth0/sync-roles.js]: good call - fixed

File: config/config.js (hunk 2 above)
Comment (asterns) [reply config/config.js]: +1 - fixed

**Pull request merged**